package.json

```diff
@@ -64,6 +64,7 @@
     "grunt-jscs": "2.1.0",
     "grunt-mocha-cli": "1.14.0",
     "hawk": "3.1.0",
+    "halfpenny": "*",
```

Comment:
1. yes, that is about how i envision it.  the person who is doing the review will only also have to link once, or, if they have _already linked_, just pull both feature branches
2. you _cannot_ publish duplicate versions to npm, so thats good.   we can have the policy of _only_ publishing on release of `release`, and publishing during the build.  In this regard our gitflow still holds.
    1. we can also make sure that we don't step on each other's toes in feature branches by rejecting commits if it does not rev the CHANGELOG.md.
    1. the publish script can publish the latest version from the changelog (if it's changed)

Great points.  These are definitely downsides of pulling the client _out_ of the API itself.

To summarize and attempt to address the concerns mentioned thus far:

1. "The dev workflow is different, because the peer-reviewer may need to perform additional steps"
    1. if the reviewer already has linked, they just need to pull both repos.  we already do this.
    1. if the reviewer has not linked, they need to pull both repos and `link`.  this is new.
        1. the good news is, if a major change occurs, the tests won't pass if linking is forgotten :)
1. "This could break the gitflow workflow"
    1. having a package depend on another package is nothing new, and has been shown to work in our many repositories using build processes to-date.  the only added complexity is how and when to publish the package we are now maintaining.
        1. we could publish halfpenny immediately on feature merge, and let nodeapi determine the version to include via package.json
            1. this is not threat.  the ~same race conditions of feature development apply here, only that whomever finishes their feature first gets publish rights to the next version #.
            1. this _may be the ideal_ option
        1. we could publish halfpenny during the build process, as suggested above